File: js/src/binary_search/binary_search.ts
export function binary_search(arr: number[], target: number): number | null {
  let low = 0;
  let high = arr.length - 1;

  while (low <= high) {
    const mid = Math.floor((low + high) / 2);
    const item = arr[mid];

    if (item === target) {
      return target;
    }

    if (target < item) {
      high = mid - 1;
    } else {
      low = mid + 1;
    }
  }

  return null;
}

File: go/binary_search/binary_search.go
package binary_search

func BinarySearch(arr []int, target int) int {
	low := 0
	high := len(arr) - 1

	for low <= high {
		mid := (low + high) / 2
		item := arr[mid]

		if item == target {
			return item
		}

		if item < target {
			low = mid + 1
		} else {
			high = mid - 1
		}
	}

	return -1
}
